src/SolutionExplorerFileSystemRepository.ts

```diff
@@ -24,6 +27,62 @@ export class SolutionExplorerFileSystemRepository implements ISolutionExplorerRe
     this._trashFolderLocation = trashFolderLocation;
   }
 
+  public watchFile(filepath: string, callback: (event: string, previousFilepath: string, newFilename: string) => void): void {
+    let isCollectingEvents: boolean = false;
+    let eventsOccured: Array<string> = [];
+
+    const watcher: fs.FSWatcher = fs.watch(filepath, async(event: string, newFilename: string) => {
+      eventsOccured.push(event);
+      if (isCollectingEvents) {
+        return;
+      }
+
+      isCollectingEvents = true;
+      await this._wait100Ms();
+
+      const occuredEvent: string = eventsOccured.includes('rename') ? 'rename' : 'change';
+
+      callback(occuredEvent, filepath, newFilename);
+
+      const fileExistsNoLonger: boolean = !fs.existsSync(filepath);
```

Comment:
Optional:
Korrekt wäre:

```suggestion
      const fileNoLongerExists: boolean = !fs.existsSync(filepath);
```